Sources/MongoSwift/MongoConnectionString.swift

```diff
@@ -1033,6 +1045,19 @@ extension StringProtocol {
         return decoded
     }
 
+    fileprivate func isIPv4() -> Bool {
```

Comment:
The regular expression for IPv4 addresses is quite unwieldy ([see](https://devblogs.microsoft.com/oldnewthing/20060522-08/?p=31113)) so I opted for this method for the sake of readability/correctness.